src/vorta/views/exclude_dialog.py

```diff
@@ -310,6 +352,73 @@ def preset_item_changed(self, item):
 
         self.populate_preview_tab()
 
+    def exclude_if_present_item_changed(self, item):
+        self.save_exclude_if_present_patterns()
+
+    def exclude_if_present_context_menu(self, pos):
+        self.create_context_menu(self.excludeIfPresentList, self.remove_pattern_exclude_if_present)
+
+    def remove_pattern_exclude_if_present(self, index=None):
+        if not index:
+            indexes = self.excludeIfPresentList.selectedIndexes()
+            for index in reversed(sorted(indexes)):
+                self.excludeIfPresentModel.removeRow(index.row())
+        else:
+            self.excludeIfPresentModel.removeRow(index.row())
+        self.save_exclude_if_present_patterns()
+
+    def add_pattern_exclude_if_present(self):
+        self.add_pattern(self.excludeIfPresentList, self.excludeIfPresentModel)
+        self.save_exclude_if_present_patterns()
+
+    def exclude_if_present_pattern_editing_finished(self, editor):
+        self.text_editing_finished(self.customExclusionsModel)
+        self.save_exclude_if_present_patterns()
+
+    def save_exclude_if_present_patterns(self):
+        '''
+        Save Exclude If Present files to profile.
+        '''
+        patterns = []
+        for row in range(self.excludeIfPresentModel.rowCount()):
+            item = self.excludeIfPresentModel.item(row)
+            text = item.text().strip()
+
+            # Remove any existing prefix
+            if text.startswith('[x] '):
+                text = text[4:]
+            elif text.startswith('[] '):
+                text = text[3:]
+
+            # Add the correct prefix based on the check state
+            prefix = '[x] ' if item.checkState() == Qt.CheckState.Checked else '[] '
+            patterns.append(prefix + text)
+
+        self.profile.exclude_if_present = '\n'.join(patterns)
+        self.profile.save()
+        self.populate_preview_tab()
+
+    def populate_exclude_if_present_patterns(self):
+        '''
+        Populate the 'Exclude If Present' list from the profile.
+        '''
+        patterns = self.profile.exclude_if_present.split('\n') if self.profile.exclude_if_present else []
+        for pattern in patterns:
+            item = QStandardItem()
+            if pattern.startswith('[x]'):
+                item.setText(pattern[4:])
+                item.setCheckable(True)
+                item.setCheckState(Qt.CheckState.Checked)
+            elif pattern.startswith('[]'):
+                item.setText(pattern[3:])
+                item.setCheckable(True)
+                item.setCheckState(Qt.CheckState.Unchecked)
+            else:
+                item.setText(pattern)
+                item.setCheckable(True)
+                item.setCheckState(Qt.CheckState.Unchecked)
```

Comment:
Doesn't this mean that all existing exclude settings the user had are now disabled?

Reply:
@m3nu That's true, I didn't take the existing ones into account. Should I try to make them checked by default?

Reply:
Yes, I would check them by default.